src/test/groovy/tree/NavigationTest.groovy

```diff
@@ -22,29 +22,37 @@ package groovy.tree
  * Simple test of tree walking
  */
 class NavigationTest extends GroovyTestCase {
-    
+
     void testDepthFirst() {
-        def tree = createTree()
-        
+        def tree = createTreeFromNodeBuilder()
         def names = tree.depthFirst().collect { it.name() }
-        def expected = ['a', 'b1', 'b2', 'c1', 'c2', 'b3', 'b4', 'c3', 'c4', 'b5']
-        
-        assert names == expected
+        assert names == ['a', 'b1', 'b2', 'c1', 'c2', 'b3', 'b4', 'c3', 'c4', 'b5']
     }
-    
+
     void testBreadthFirst() {
-        def tree = createTree()
-        
+        def tree = createTreeFromNodeBuilder()
         def names = tree.breadthFirst().collect { it.name() }
-        def expected = ['a', 'b1', 'b2', 'b3', 'b4', 'b5', 'c1', 'c2', 'c3', 'c4']
-        
-        assert names == expected
+        assert names == ['a', 'b1', 'b2', 'b3', 'b4', 'b5', 'c1', 'c2', 'c3', 'c4']
+    }
+
+    void testPrePostOrder() {
+        def root = createTreeFromXmlParser()
+        def combos = [[false, true], ['depthFirst', 'breadthFirst']].combinations()
+        def actual = combos.collect{ preorder, type ->
+            root."$type"(preorder)*.name()
+        }*.toString().join('\n')
+        def expected = '''
+            [childa, childb, parent1, childa, grandchild, childb, childc, parent2, root]
```

Comment:
Minor point, but might improve test readability by changing from a multiline string to a concatenated string with trailing comments labeling the combination.